packages/expo-cli/src/credentials/views/IosPushCredentials.ts

```diff
@@ -38,6 +38,11 @@ export class CreateIosPush implements IView {
     log('Successfully created Push Notification Key\n');
     displayIosUserCredentials(pushKey);
     log();
+
```

Comment:
This action is called from other places e.g. as part of the setup before build where this behaviour would not be correct.

This was intended to simple CRUD action that results in the creation of a new push key, if the intention is that ensuring valid push key is assigned to the app then logic for that should be different and is already present in build command.

If it's unclear that this command won't assign dist cert to the app then maybe it's better to add some warnings. If you want to modify this action without affecting build then create new action that is using `CreateIosPush` class.
```
async open(ctx: Context): Promise<> {
  await runCredentialsManager(ctx, new CreateIosPush(accountName));
  // some code you want to add
}
```
and replace it here https://github.com/expo/expo-cli/blob/master/packages/expo-cli/src/credentials/views/Select.ts#L99

Reply:
ah I see, thanks for catching that

Changed it and created a new class that extends `CreateIosPush` instead - https://github.com/expo/expo-cli/pull/3098/commits/dcae9ca0983a5b9fc7c136fbbc056f9d79c56088